server/main.cpp

```diff
@@ -80,28 +82,32 @@ int main(int argc, char* argv[])
 		}
 	}
 
-	std::clog << "Opening server port at " << port << '\n';
+	std::clog << "Opening server port at " << commandPort << '\n';
+	std::clog << "Listening for events on port " << eventPort << '\n';
 
 	try {
-		Socket server(port);
+		Socket commandServer(commandPort);
+		Socket eventServer(eventPort);
 
 		bool running = true;
 		do {
 			try {
-				Socket client = server.accept();
-				std::clog << "Incoming connection from " << client.getPeerName().data << '\n';
+				Socket commandClient = commandServer.accept();
+				Socket eventClient = eventServer.accept();
```

Comment:
I think this handshake method isn't stable. We should negotiate the event connection after the initial command connection was done. As in, the server thread/fork would tell the client a port to which establish an Event connection.

I already had a patch half-done to multiplex the one connection, so hopefully I will be able to merge in your work and we might end up not needing the two sockets going forward; so for now, this is acceptable.

Reply:
Note that the method you're using happens before versions checks (on the handshake), so if server/client versions are mixed, the server will likely stall waiting for the event connection.